Unittest of Config module is broken

### Checklist
- [X] Bug from Service (Vatz)
- [ ] Bug from Plugin (Vatz) 
- [ ] Malfunction on CI/CD 
- [ ] etc(e.g, Hardware version doesn't match, )

---

### **Describe the bug**
> A clear and concise description of what the bug is.

Config module test is broken.

### **How to Reproduce**
> Steps to reproduce the behavior:

```
rootwarp@Mindhack2021 ~/g/d/vatz main > make coverage
echo "Test Coverage script will be here"
Test Coverage script will be here
ok      github.com/dsrvlabs/vatz        1.779s  coverage: 11.0% of statements
ok      github.com/dsrvlabs/vatz/manager/api    0.374s  coverage: 0.0% of statements [no tests to run]
{"level":"error","module":"config","time":"2022-10-14T19:21:56+09:00","message":"loadConfig Error: open not_existing_file.yaml: no such file or directory"}
{"level":"error","module":"config","time":"2022-10-14T19:21:56+09:00","message":"Please, initialize VATZ with command `./vatz init` to create config file `default.yaml` first or set appropriate path for config file `default.yaml`."}
FAIL    github.com/dsrvlabs/vatz/manager/config 0.379s
ok      github.com/dsrvlabs/vatz/manager/dispatcher     0.672s  coverage: 8.9% of statements
ok      github.com/dsrvlabs/vatz/manager/executor       0.879s  coverage: 68.2% of statements
ok      github.com/dsrvlabs/vatz/manager/healthcheck    1.302s  coverage: 45.0% of statements
?       github.com/dsrvlabs/vatz/manager/types  [no test files]
?       github.com/dsrvlabs/vatz/mocks  [no test files]
ok      github.com/dsrvlabs/vatz/rpc    2.541s  coverage: 67.2% of statements
FAIL
make: *** [coverage] Error 1
```

### **Expected behavior**
> A clear and concise description of what you expected to happen.

Tests passed.

### **Screenshots**
> If applicable, add screenshots to help explain your problem.

--- a/cmd.go
+++ b/cmd.go
@@ -2,6 +2,7 @@ package main
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"net/http"
@@ -123,7 +124,15 @@ func createStartCommand() *cobra.Command {
 			log.Info().Str("module", "main").Msgf("load config %s", configFile)
 			log.Info().Str("module", "main").Msgf("logfile %s", logfile)
 
-			config.InitConfig(configFile)
+			_, err := config.InitConfig(configFile)
+			if err != nil {
+				log.Error().Str("module", "config").Msgf("loadConfig Error: %s", err)
+				if errors.Is(err, os.ErrNotExist) {
+					msg := "Please, initialize VATZ with command `./vatz init` to create config file `default.yaml` first or set appropriate path for config file default.yaml."
+					log.Error().Str("module", "config").Msg(msg)
+				}
+				return err
+			}
 
 			ch := make(chan os.Signal, 1)
 			return initiateServer(ch)

--- a/manager/config/config.go
+++ b/manager/config/config.go
@@ -5,7 +5,6 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
-	"os"
 	"strings"
 	"sync"
 
@@ -153,28 +152,42 @@ func (p *parser) overrideDefault(config *Config) {
 }
 
 // InitConfig - initializes VATZ config.
-func InitConfig(configFile string) *Config {
+func InitConfig(configFile string) (*Config, error) {
+	if vatzConfig != nil {
+		log.Info().Str("module", "config").Msgf("Config already loaded")
+		return vatzConfig, nil
+	}
+
+	var configError error
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
 	configOnce.Do(func() {
+		defer wg.Done()
+
+		log.Error().Str("module", "config").Msgf("Load Config %s", configFile)
+
+		var (
+			configData []byte
+		)
+
 		p := parser{}
-		configData, err := p.loadConfigFile(configFile)
-		if err != nil {
-			if strings.Contains(err.Error(), "no such file or directory") {
-				log.Error().Str("module", "config").Msgf("loadConfig Error: %s", err)
-				log.Error().Str("module", "config").Msg("Please, initialize VATZ with command `./vatz init` to create config file `default.yaml` first or set appropriate path for config file `default.yaml`.")
-				os.Exit(1)
-			}
-			panic(err)
-		}
-		config, err := p.parseYAML(configData)
-		if err != nil {
-			log.Error().Str("module", "config").Msgf("parseYAML Error: %s", err)
-			panic(err)
+		configData, configError = p.loadConfigFile(configFile)
+		if configError != nil {
+			return
 		}
 
-		vatzConfig = config
+		vatzConfig, configError = p.parseYAML(configData)
+		if configError != nil {
+			log.Error().Str("module", "config").Msgf("parseYAML Error: %s", configError)
+			return
+		}
 	})
 
-	return vatzConfig
+	wg.Wait()
+
+	return vatzConfig, configError
 }
 
 // GetConfig returns current Vatz config.

--- a/manager/config/config_test.go
+++ b/manager/config/config_test.go
@@ -1,6 +1,7 @@
 package config
 
 import (
+	"errors"
 	"net/http"
 	"os"
 	"sync"
@@ -268,32 +269,28 @@ func TestURLConfig(t *testing.T) {
 }
 
 func TestNotExistConfigFile(t *testing.T) {
-	defer func() {
-		recover()
-	}()
+	configOnce = &sync.Once{} // Overrice Once
+	vatzConfig = nil
 
-	_ = InitConfig("not_existing_file.yaml")
+	cfg, err := InitConfig("not_existing_file.yaml")
 
-	// DO NOT REACH HERE
-	t.Error("no panic occures")
+	assert.Nil(t, cfg)
+	assert.NotNil(t, err)
+	assert.True(t, errors.Is(err, os.ErrNotExist))
 }
 
 func TestInvalidYAMLFormat(t *testing.T) {
-	defer func() {
-		recover()
-	}()
-
 	configOnce = &sync.Once{} // Overrice Once
+	vatzConfig = nil
 
 	f, err := createDefaultConfigFile(configInvalidYAMLContents)
 	defer os.Remove(f.Name())
 
 	assert.Nil(t, err)
 
-	_ = InitConfig(DefaultConfigFile)
+	_, err = InitConfig(DefaultConfigFile)
 
-	// DO NOT REACH HERE
-	t.Errorf("no panic")
+	assert.NotNil(t, err)
 }
 
 func createDefaultConfigFile(contents string) (*os.File, error) {
@@ -370,9 +367,10 @@ func TestGetConfig(t *testing.T) {
 	assert.Nil(t, err)
 
 	// Init Config.
-	cfg := InitConfig(DefaultConfigFile)
+	cfg, err := InitConfig(DefaultConfigFile)
 
 	// Asserts.
+	assert.Nil(t, err)
 	assert.Equal(t, test.ExpectProtocolID, cfg.Vatz.ProtocolIdentifier)
 	assert.Equal(t, test.ExpectVatzPort, cfg.Vatz.Port)
 	assert.Equal(t, test.ExpectHostName, cfg.Vatz.NotificationInfo.HostName)

